api/envoy/config/core/v3/health_check.proto

```diff
@@ -103,8 +103,20 @@ message HealthCheck {
     // [#not-implemented-hide:] HTTP specific payload.
     Payload send = 3;
 
-    // [#not-implemented-hide:] HTTP specific response.
-    Payload receive = 4;
+    // Specifies a list of HTTP expected responses to match in the first ``response_buffer_size`` bytes of the response body.
+    // If it is set, both the expected response check and status code determine the health check.
+    // When checking the response, “fuzzy” matching is performed such that each payload block must be found,
+    // and in the order specified, but not necessarily contiguous.
+    //
+    // .. note::
+    //
+    //   It is recommended to set ``response_buffer_size`` based on the Payload size for efficiency,
+    //   otherwise all response body will be recorded.
+    repeated Payload receive = 4;
+
+    // Specifies the size of response buffer in bytes that is used to Payload match.
+    // All the response body will be used for matching if it is not set and ``receive`` is not empty.
```

Comment:
@htuch the default value can be "safe" (e.g., set max-size to 1024), or it could be unbounded (to keep parity with tcp). WDYT?

@botengyao please add what will happen when the value is set to 0.

Reply:
Thank you! 

The current approach is if it is not set or 0 it will be unbounded. Also put a comment above to say it is recommended to set the buffer size.

Or we can set the default value is 1024 when it is not set, and to use unbound by setting to 0. 

Both should work.

Reply:
Prefer sensible default bound as client-specific value like 1024 when not set. When set, use the set value, 0 as unlimited.

Reply:
Thank you, done!